### DISCUSSION TITLE
Laser skips some cuts

### DISCUSSION THREAD

**Discussion by umvegas** [arkypita/LaserGRBL#2403] [Q&A] [answered]
The code below was meant to cut three circles. The first two cut as expected, but the third one does not. The laser travels to the position, but then doesn't move through circle shape.

I have included comments to indicate what I think is happening. I must be misunderstanding something.

What am I doing wrong?

```
M3 S0
S0                         ; laser off
G0X72.12Y21.314            ; move to the initial position
S150                       ; laser power 15%
G3X72.12Y21.314I-6J0F1000  ; move through a circle, 6mm radius
S0                         ; laser off
G0X265.119Y54.5            ; move to new position
S150                       ; laser power 15%
G3X265.119Y54.5I-6J0F1000  ; draw the circle, again 6mm radius
S0                         ; laser off
G0X72.12Y55.982            ; move to another position
S150                       ; laser power 15%
G3X72.12Y55.982I-6J0F1000  ; draw another 6mm circle, this line seems to have no effect
S0                         ; laser off
M5 S0
```

**Comment by Hostaler** [comment]
The program is fine. Restart your machine and reload it. It would be nice to add M02 at the end of the program or M30

**Reply by umvegas** [reply]
I have restarted the machine many times, and even regenerated the gcode with LaserGRBL or written it by hand. Regardless, the machine does not cut the third circle. The complete cut includes shapes cut before these holes and other shapes cut after, all of which are cut as expected. Whether those shapes are included or not, the third hole is skipped.

What is the meaning of M02 and M30? I've looked these codes up, but different sources say different things.

Thanks.

**Reply by arkypita** [reply]
What machine?

**Reply by arkypita** [reply]
I have seen issues with machine skipping some circles in the past with Algo, solved by a firmware update

**Reply by umvegas** [reply]
It is an Algo. I will check for updates and try again.

**Reply by Hostaler** [reply]
> He reiniciado la máquina muchas veces e incluso he regenerado el código G con LaserGRBL o lo he escrito a mano. De todas formas, la máquina no corta el tercer círculo. El corte completo incluye formas cortadas antes de estos agujeros y otras formas cortadas después, todas las cuales se cortan como se esperaba. Ya sea que se incluyan esas formas o no, se omite el tercer agujero.
> 
> ¿Cuál es el significado de M02 y M30? He buscado estos códigos, pero distintas fuentes dicen cosas distintas.
> 
> Gracias.

M02 es **fin** de programa sin volver al principio del programa y M30 es **fin** de programa y vuelve al principio del programa y se queda en espera.

**Reply by umvegas** [reply]
> I have seen issues with machine skipping some circles in the past with Algo, solved by a firmware update

I checked for a firmware update, but my machine is already up-to-date. I re-generated the gcode and still the same circle is skipped. The machine goes to the correct starting point, but it draws no circle, instead moving on to the next circle, which is drawn correctly, as is every other shape in the gcode.

**Reply by umvegas** [reply]
Here's a solution, but I don't know why it works:
I changed the Y coordinate from 55.982 to 55.98. That's it, and now it works. How can that make sense?

**Comment by Hostaler** [comment]
I think it's because your motherboard couldn't read thousandths.

**Reply by umvegas** [reply]
Sounds plausible, except that thousandths appear elsewhere in the file and don't cause problems.

**Reply by Hostaler** [reply, marked as answer]
I already saw it, it will be another mystery of technology.
